fluentd/generate_throttle_configs.rb

```diff
@@ -159,7 +159,7 @@ def create_default_docker(excluded)
   pos_file "#{ENV['JSON_FILE_POS_FILE'] || '/var/log/es-containers.log.pos'}"
   time_format %Y-%m-%dT%H:%M:%S.%N%Z
   tag kubernetes.*
-  format json
+  format #{ENV['USE_CRIO'] == 'true' ? '/^(?<time>.+) (?<stream>stdout|stderr)( (?<logtag>.))? (?<message>.*)$/' : 'json'}
   keep_time_key true
   read_from_head "#{ENV['JSON_FILE_READ_FROM_HEAD'] || 'true'}"
```

Comment:
`JSON_FILE_READ_FROM_HEAD` with CRI-O, is that confusing?

Reply:
@portante if you find it confusing, others will too, so we'll need to have corresponding CRI-O options for every JSON_FILE option.  You'll also need to document the parameters in `docs/log-drivers-and-throttling.md`

Reply:
@richm @portante given we never exposed `JSON_FILE_POS_FILE` and other `JSON_*` config knobs to openshift-ansible, would it be possible to rename and reuse the current `JSON_*` env variables for both docker json-file driver as well as cri-o or we want to keep the old names and mirror the settings with `CRIO_*` env variables anyway?

Both log to `/var/log/containers/*` with same naming conventions, only the log line format is different.

Reply:
We never exposed these parameters to ansible because we don't want users to be able to easily set them.  However, it is useful for dev/qe to be able to set them for testing purposes.

If we decide to go with CRI-O options, then the default value should be the same as the value of the corresponding JSON_FILE value e.g. 
```
CRIO_FILE_PATH=${CRIO_FILE_PATH:-${JSON_FILE_PATH:-'/var/log/containers/*.log'}}
CRIO_FILE_POS_FILE=${CRIO_FILE_POS_FILE:-${JSON_FILE_POS_FILE:-'/var/log/es-containers.log.pos'}}
```
etc.

Reply:
and is it preferable to have both `CRIO_*` and `JSON_*` and not unify the settings to something like `LOG_*`?

I could imagine, given these config knobs are exposed to QE and ourselves only, renaming the original env variables from `$JSON_FILE_PATH` to `$LOG_FILE_PATH` won't necessarily do too much harm and would mean we don't double the count of env variables related to these settings. As far as I can tell, we have 3 - `JSON_FILE_PATH`,`JSON_FILE_POS_FILE`, and `JSON_FILE_READ_FROM_HEAD` so that would go up to 6, which is probably also not such a big deal.

Reply:
`LOG_*` is too generic - could also apply to journald logs.  Something like `CONTAINER_LOG_FILE_*` would be suitable, if a bit verbose.  @portante any suggestions?

Reply:
We might want to align the names to the container subsystem in use, like docker and crio.

Reply:
I don't want to change the existing options.  I think it is fairly obvious that JSON_FILE == docker.

Reply:
if we need to handle heterogeneous environments with both cri-o logs and json docker logs at the same time on the same node, then I'm afraid we won't be able to avoid `multi_format` input plugin. At the same time, probably worth mentioning that with `multi_format` [we lose the ability to concatenate multi-line logs natively](https://github.com/repeatedly/fluent-plugin-multi-format-parser) by fluentd's `multiline` parser once cri-o has that fully working and we may have to write our own input plugin for that.

Reply:
I think we need some input from @mrunalp about how users will upgrade from docker to crio.

If we cannot support multiple formats at the same time, what is our upgrade story?

- user must make sure that all /var/log/containers/*.log files are read

How?

- user must make sure /var/log/containers/*.log (and /var/lib/docker/containers?) is empty

How?

Reply:
I don't think it is user's responsibility, it is kubelet who maintains symlinks in `/var/log/containers/*` so as far as we know how it is going to behave and if there are going to be nodes with heterogeneous container runtime environment, we can make a decision

https://github.com/kubernetes/kubernetes/blob/244549f02afabc5be23fc56e86a60e5b36838828/pkg/kubelet/kubelet.go#L1282-L1287

Reply:
added `CRIO_*` env vars to complement the already existing `JSON_*` env vars